WalletWasabi/Blockchain/Transactions/TransactionFactory.cs

```diff
@@ -356,6 +359,24 @@ private void UpdatePSBTInfo(PSBT psbt, SmartCoin[] spentCoins, HdPubKey changeHd
 					psbt.AddKeyPath(changeHdPubKey.PubKey, rootKeyPath, changeHdPubKey.P2wpkhScript);
 				}
 			}
+
+			if (Store is null)
+			{
+				return;
+			}
+
+			foreach (var input in spentCoins)
+			{
+				var coinInputTxID = input.TransactionId;
+				if (Store.TransactionStore.TryGetTransaction(coinInputTxID, out var txn))
+				{
+					var psbtInput = psbt.Inputs.FirstOrDefault(x => x.PrevOut.Hash == coinInputTxID);
+					if (psbtInput != null)
+					{
+						psbtInput.NonWitnessUtxo = txn.Transaction;
```

Comment:
`psbtInput.NonWitnessUtxo = txn.Transaction;`

UTXO is a UTXO, a tx is a tx. What's going on here?

Reply:
NonWitness UTXO has Transaction as its type so the queried txn is placed in there...

Reply:
This is correct.

Reply:
if the multiple inputs from the same tx then add the tx to everywhere